pytorch_lightning/core/lightning.py

```diff
@@ -405,7 +405,14 @@ def write_prediction(
             when running in distributed mode, calling ``write_prediction`` will create a file for
             each device with respective names: ``filename_rank_0.pt``, ``filename_rank_1.pt``, ...
 
+        .. deprecated::v1.3
+            Will be removed in v1.5.0.
         """
+        rank_zero_deprecation(
```

Comment:
you can also use a deprecation wrapper as we have in metrics...
https://borda.github.io/pyDeprecate/